omnigibson/utils/grasping_planning_utils.py

```diff
@@ -0,0 +1,383 @@
+#     towards_object_in_world_frame = bbox_center_in_world - grasp_center_pos
+#     towards_object_in_world_frame /= np.linalg.norm(towards_object_in_world_frame)
+
+#     # For the grasp, we want the X+ direction to be the direction of the object's surface.
+#     # The other two directions can be randomized.
+#     rand_vec = np.random.rand(3)
+#     rand_vec /= np.linalg.norm(rand_vec)
+#     grasp_x = towards_object_in_world_frame
+#     grasp_y = np.cross(rand_vec, grasp_x)
+#     grasp_y /= np.linalg.norm(grasp_y)
+#     grasp_z = np.cross(grasp_x, grasp_y)
+#     grasp_z /= np.linalg.norm(grasp_z)
+#     grasp_mat = np.array([grasp_x, grasp_y, grasp_z]).T
+#     grasp_quat = Rotation.from_matrix(grasp_mat).as_quat()
+
+#     grasp_pose = (grasp_center_pos, grasp_quat)
+#     grasp_candidate = [(grasp_pose, towards_object_in_world_frame)]
+
+#     return grasp_candidate
+
+def get_grasp_position_for_open(robot, target_obj, should_open, relevant_joint=None, num_waypoints="default"):
+    # Pick a moving link of the object.
+    relevant_joints = [relevant_joint] if relevant_joint is not None else _get_relevant_joints(target_obj)[1]
+    if len(relevant_joints) == 0:
+        raise ValueError("Cannot open/close object without relevant joints.")
+
+    # Make sure what we got is an appropriately open/close joint.
+    random.shuffle(relevant_joints)
+    selected_joint = None
+    for joint in relevant_joints:
+        # from IPython import embed; embed()
+        # if joint.name == "drawer:joint_j_link_2":
+        #     selected_joint = joint
+        #     break
+        current_position = joint.get_state()[0][0]
+        joint_range = joint.upper_limit - joint.lower_limit
+        openness_fraction = (current_position - joint.lower_limit) / joint_range
+        if (should_open and openness_fraction < 0.8) or (not should_open and openness_fraction > 0.05):
+            selected_joint = joint
+            break
+        # current_position = joint.get_state()[0][0]
+        # joint_range = joint.upper_limit - joint.lower_limit
+        # openness_fraction = (current_position - joint.lower_limit) / joint_range
+        # if (should_open and openness_fraction < 0.8) or (not should_open and openness_fraction > 0.05):
+        #     selected_joint = joint
+
+    if selected_joint is None:
+        return None
+
+    if selected_joint.joint_type == JointType.JOINT_REVOLUTE:
+        return grasp_position_for_open_on_revolute_joint(robot, target_obj, selected_joint, should_open, num_waypoints=num_waypoints)
+    elif selected_joint.joint_type == JointType.JOINT_PRISMATIC:
+        return grasp_position_for_open_on_prismatic_joint(robot, target_obj, selected_joint, should_open, num_waypoints=num_waypoints)
+    else:
+        raise ValueError("Unknown joint type encountered while generating joint position.")
+    
+
+def grasp_position_for_open_on_prismatic_joint(robot, target_obj, relevant_joint, should_open, num_waypoints="default"):
```

Comment:
this is really cool! again, would be helpful to have docstring so we can get an idea (high level) how it's supposed to behave. this would also help us debug too, if we know sort of "what" to expect

Reply:
I'm putting in a simple docstring for now but we'll document this later when we've updated it to a more robust approach. This opening/closing leaves much to be desired currently.